python/tank/platform/bundle.py

```diff
@@ -663,11 +663,49 @@ def create_hook_instance(self, hook_expression):
 
         .. note:: For more information about hook syntax, see :class:`~sgtk.Hook`
 
+        An optional `base_class` can be provided to override the default ``Hook``
+        base class. This is useful for bundles that create hook instances at
+        execution time and wish to provide default implementation without the need
+        to configure the base hook.
```

Comment:
maybe worth pointing our here too that the custom base class needs to derive from hook?

Reply:
done.